### DISCUSSION TITLE
How are efficiency performance and efficiency score calculated?

### DISCUSSION THREAD

**Discussion by DennisAdenHollander** [TUE-ARIA/NNCV#38] [General]
I have uploaded the first submission for the efficiency, but i receive an efficiency performance of 0.000. I was wondering on what basis the performance and score is calculated also for my research paper

**Comment by TimJaspers0801** [comment]
The Efficiency score is based on the number of Gflops of your model (you want to minimize this value). The Efficiency performance is calculated as the mean Dice score divided by the number of Gflops. However, there is an additional condition: your model must achieve at least 80% of the baseline performance (TA_baseline) to receive an efficiency score. If it falls below this threshold, the efficiency performance is set to 0.000. This is also the reason why you received a performance of 0.000. That said, don’t worry—this challenge is designed to be difficult, and achieving a good efficiency score is not necessarily easy. Keep working on it, and let us know if you need any guidance!"

**Comment by kmilas** [comment]
So the efficiency score does not take into account model parameters, inference speed or memory consumption ? Also I use two tools to calculate the flops of my model and get different results. For example with thop https://github.com/ultralytics/thop i get 10.68 GMACs and from flops counter https://github.com/sovrasov/flops-counter.pytorch I get 13.36 GMACs. Also we need to care only about gpu inference or also cpu inference and  can we know the  image resolution that you use for testing ?

**Reply by TimJaspers0801** [reply]
You're right to raise these points. While model parameters, inference speed, and memory consumption are all important practical metrics, they are often highly correlated—especially with the number of FLOPs. For this challenge, we consider the number of FLOPs to be the most representative and comparable measure of efficiency across submissions. 

The image resolution used for testing is equal to the resolution you use for training your model. In the process_data.py file, you have access to the preprocess function. Here, you can simply resize the images to the intended resolution.

**Comment by DennisAdenHollander** [comment]
I do not get how my more accurate model (trained using a combined loss function of CE loss and DICE loss) gets a lower mean dice score than the worst model in my arsenal? I mean even when I check the validation set the standard model performs very poorly and the secondary model performs decent, however the score does not represent this at all...

**Reply by TimJaspers0801** [reply]
Let me check. What is your username on Codalab? If your model still performs worse than 80% of our Baseline_TA model, you will get a score of 0.0. What are the mean_dices scores you're getting?

**Reply by DennisAdenHollander** [reply]
My username is Dennis2000. Its not necessarily the 0.0 score that is bugging me, even though my models reaches a validation score (and mean dice score) similar to the TA_baseline mean dice score according to wandb. It is more that I have now uploaded the most accurate model I have built (I save and train every version), which is a lot more accurate than my standard model which I submitted first (and second), which gets a lower dice score. I cannot make sense of it. I have attached their validation scores from the training so you can see what confuses me. Somehow, the purple model is better than the gray one (while the loss is meant to be minimized). I hope you can help me understand it, since I am either making a huge mistake somewhere or lack some understanding :)

valid_dice = dice_loss = 1-dice score
valid_loss = 0.5*ce+0.5*dice
![W B Chart 08_04_2025, 17_19_42](https://github.com/user-attachments/assets/28a4de5d-1c50-42cd-b9dc-0b7dd32c5284)
![W B Chart 08_04_2025, 17_19_15](https://github.com/user-attachments/assets/2d51afe2-31a2-4046-b7da-3be16813f820)

The best models dice score on my validation set is 0.495, which is higher than 0.468.

**Reply by TimJaspers0801** [reply]
Sorry, I do not fully get your problem yet.

Have you submitted both the purple and gray models to Codalab? What were the corresponding mean dice scores?

As I can see, your best performance on CodaLab is quite low (mean dice=0.183) compared to the performance on your own validation set. Maybe something is going wrong in the preprocess function?

**Reply by DennisAdenHollander** [reply]
My purple model got a mean dice of 0.2 on codalab previously, the gray model gets 0.184, which is even lower while the model should be a lot better, and my problem was how i do not get how that is possible.

So purple model: 0.204 i think
Gray: 0.184

I have not adapted the preprocess function i think. I left process_data.py as is

**Reply by TimJaspers0801** [reply]
Yes, I expect that this is the problem. You need to adjust the preprocess function in a way that it returns an output that your model expects. So, I don't know at what resolution your model is trained or what normalization you do, but this could very well impact the model's performance.

**Reply by DennisAdenHollander** [reply]
Where can i find this function?

**Reply by TimJaspers0801** [reply]
inside the process_data.py file

**Reply by DennisAdenHollander** [reply]
And how should it look like, I only resize to 256,256

**Reply by TimJaspers0801** [reply]
Don't you perform any normalization? than you only need to resize to the correct resolution.

**Reply by DennisAdenHollander** [reply]
i have only changed figure size as was instructed, for the rest I have not edited this file at all...

**Reply by TimJaspers0801** [reply]
no, except if you came up with something special in the preprocessing of the data.

**Reply by DennisAdenHollander** [reply]
I have not done this. I have compared the original files with my current ones and the only difference is:

img = transforms.functional.resize(img, size=(256, 256), interpolation=transforms.InterpolationMode.LANCZOS)
inside preprocess(img):

For the rest I noticed a discrepancy in the competition instructions, where it states I have to deliver preprocess.py instead of process_data.py.

Next to that I noticed my models getting high scores on codalab for dice 1, 2 and 10, where the other dices are super low (approaching zero), does this mean anything (it getting the number of classes wrong)? My model is checked next to the base UNet model, and sends out the same output shape atleast.

There is one difference between the UNet and my model, my model states n_classes = 19 whereas the UNet states n_classes = 1

Sorry that I take so much time, but I cannot check whether I fixed the issue due to me throwing away submissions and training coins for this.

**Reply by TimJaspers0801** [reply]
No worries at all, that’s exactly what this discussion page is for!

We’re aware of the discrepancy in the competition instructions; someone else has already pointed it out. For now, just stick with the process_data.py file, which includes the correct preprocess function.

From what I can see, it looks like you're on the right track. Don’t worry too much about the number of submissions—that limit is mainly there to prevent the server from getting overloaded. When you get a decent score, I can remove previous submissions for you, and your number of submissions will be back to 1.

That said, it’s a bit tricky for me to pinpoint the exact issue without seeing more details. If you have some time, feel free to drop by our office (Flux 5.092), and we can take a look at your code together.
